### PR TITLE
Feature/tutorial type

### PR DESCRIPTION
**Pull request by mikeymike**
Adds new functionality for a tutorial mode forcing users of these types of workshops to complete exercises one by one. 

Created types so that maybe we can extend in the future in some form, e.g. timed etc

### CODE REVIEW

File: app/config.php [hunk 1]
```diff
@@ -58,7 +58,8 @@
 
 return [
     'appName' => basename($_SERVER['argv'][0]),
-    ExerciseDispatcher::class => function (ContainerInterface $c) {
+    WorkshopType::class => WorkshopType::STANDARD(),
+    ExerciseDispatcher::class => factory(function (ContainerInterface $c) {
```
Comment (AydinHassan) [inline app/config.php]: any reason you added this? - it doesn't actually need it

File: src/Factory/MenuFactory.php [hunk 2]
```diff
@@ -44,13 +46,19 @@ public function __invoke(ContainerInterface $c)
...
+            ->addItems(
+                array_map(function (ExerciseInterface $exercise) use ($exerciseRenderer, $userState, $workshopType) {
+                    $isCurrent  = $exercise->getName() === $userState->getCurrentExercise();
+                    $isComplete = in_array($exercise->getName(), $userState->getCompletedExercises());
+
+                    return [
+                        $exercise->getName(),
+                        $exerciseRenderer,
+                        $userState->completedExercise($exercise->getName()),
+                        $workshopType == WorkshopType::TUTORIAL() && !$isCurrent && !$isComplete
```
Comment (AydinHassan) [inline src/Factory/MenuFactory.php]: I think I prefer the check like `$workshopType->getValue() === WorkshopType:: TUTORIAL`, I just can't get along with the `==` checks :(

Comment (AydinHassan) [inline src/Factory/MenuFactory.php]: Also can you extract this check to a method, it's not really obvious what it's doing

File: src/WorkshopType.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * @method static WorkshopType STANDARD()
+ * @method static WorkshopType TUTORIAL()
+ * @author Michael Woodward <mikeymike.mw@gmail.com>
+ */
+class WorkshopType extends Enum
+{
+    const STANDARD = 1;
+    const TUTORIAL = 2;
```
Comment (AydinHassan) [inline src/WorkshopType.php]: You can actually add methods here if you wanted like `isTutorialMode()` then you can just call that up above

**Comment by mikeymike** [comment]
hmmm should probably add some tests for this specifically I guess

File: app/config.php (hunk 1 above)
Comment (mikeymike) [inline app/config.php]: just thought it would be better to be explicit about the type by default

File: src/Factory/MenuFactory.php (hunk 2 above)
Comment (mikeymike) [inline src/Factory/MenuFactory.php]: yeah good shout not sure why I didn't just do that anyway

File: src/WorkshopType.php (hunk 3 above)
Comment (mikeymike) [inline src/WorkshopType.php]: ahh sweet would be a lot more expressive 👍

File: app/config.php (hunk 1 above)
Comment (AydinHassan) [inline app/config.php]: Just asked cuz i was removing as I found them 😂

Comment (mikeymike) [inline app/config.php]: oh wait... sorry that's something else

Comment (mikeymike) [inline app/config.php]: how the fuck

**Comment by AydinHassan** [comment]
@mikeymike can haz tests?

**Pull request merged**